src/internal/pjsdb/queue_iterator.go

```diff
@@ -0,0 +1,74 @@
+package pjsdb
+
+import (
+	"context"
+
+	"github.com/jmoiron/sqlx"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+	"github.com/pachyderm/pachyderm/v2/src/internal/stream"
+)
+
+// TODO(Fahad): add queue filter once filter is designed.
```

Comment:
TODO

Reply:
I was under the impression that its okay to have TODOs in code. Especially since this is the first iteration and we're going to be adding to this.